### PR TITLE
*: resolve several compile warnings and ci errors

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: tests/raftstore/test_multi.rs [hunk 1]
```diff
@@ -104,6 +106,10 @@ fn test_multi_cluster_restart<T: Simulator>(cluster: &mut Cluster<T>) {
     assert_eq!(cluster.get(key), Some(value.to_vec()));
 
     cluster.shutdown();
+
+    // avoid TIMEWAIT
+    thread::sleep(Duration::from_millis(500));
```
Comment (siddontang) [inline tests/raftstore/test_multi.rs]: you can use util sleep_ms directly.

File: tests/raftstore/cluster.rs [hunk 2]
```diff
@@ -107,7 +107,15 @@ impl<T: Simulator> Cluster<T> {
     }
 
     pub fn start(&mut self) {
-        self.start_with_strategy(vec![]);
+        if self.engines.is_empty() {
+            self.start_with_strategy(vec![]);
+        } else {
+            // recover from last shutdown.
+            let node_ids: Vec<u64> = self.engines.keys().cloned().collect();
```
Comment (siddontang) [inline tests/raftstore/cluster.rs]: do we call start multi times for a test case?

Comment (BusyJay) [inline tests/raftstore/cluster.rs]: Yes, `test_multi_cluster_restart`.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**